new-frontiers-14/frontier-station-14
# Add Pious trait, blessing water/blood requires faith

## About the PR



Adds a new trait, Pious, to get a bible and the ability to use it at round start.  For balance, Pious also includes the pacified trait.

The PR adds a (currently unused) "faith implanter" to get the BibleUserComponent.  This is intended to ship in mail along with those bibles, and provides players with a way to opt into Chaplain roleplay mid-round.

Blessing water into holy water or blood into wine now requires the BibleUserComponent.  Pious characters will be able to do this on round start with their equipped bible.

When a user is not Pious, they will get a notice when trying to summon familiars or bless water into wine.

### Potential Issues

I do not like how the order of operations is enforced here with the before parameter in event handler registration, and I don't like how temporary data is being stored in the BibleComponent.  This current solution is more fragile than I'd like, but I don't think I can do it with fewer changes.  Happy to hear advice to the contrary.

## Why / Balance

Removal of chaplain role made this difficult to get outside of admemes.  This should be useful for better chaplain RP, and I'd like to throw the faith implanter in mail packages with the bible if you get one.

The pop-up messages for the failure to bless or summon with a bible were added for extra feedback.

The holy water change seems fitting to the role, and should result in fewer impromptu miracles on station from the unwashed masses.

## How to test


Build and run locally.
1. Check the trait list.  Piety should show up with a sensible description.
2. Spawn in as a Pious character.  You should have a bible, and the bible should work as usual (see steps 9-11).
3. Run /golobby, spawn in as a non-Pious character.
4. Spawn a faith implanter, a bottle of water, and a bible.
5. Try to use the bible to pray, you should get a notice that you aren't worthy.
6. Try to use the bible to bless the water, you should get a notice that you aren't worthy.
7. Try to use the bible to summon a familiar, you should get a notice that you aren't worthy.
8. Inject the faith implanter.
9. Use the bible to pray.  It should work.
10. Use the bible to bless the water.  It should work.
11. Use the bible to summon a familiar, it should work.

## Media


Traits:
![image](https://github.com/new-frontiers-14/frontier-station-14/assets/166147148/62429f5b-f21d-4f06-b94e-67e632a84b2a)

Faith implanter in use (note: prototype exists, but is not used in-game):
https://github.com/new-frontiers-14/frontier-station-14/assets/166147148/e8a2c81f-8f0e-4f7c-9131-c2023cc30528


## Breaking changes


Adds `BlockMix`, `LastInteractingUser` public fields to `BibleComponent`.

**Changelog**



:cl:
- add: Characters can now select the Pious trait to start with a bible, the ability to use it, and a vow of pacifism.
- tweak: Water and blood can only be blessed with a bible if the user is Pious.
